site/gatsby-site/src/components/forms/IncidentReportForm.js

```diff
@@ -108,6 +108,16 @@ const IncidentReportForm = () => {
 
   const [parsingNews, setParsingNews] = useState(false);
 
+  useEffect(() => {
+    try {
+      const url = new URL(values?.url);
+
+      setFieldValue('source_domain', url.hostname);
+    } catch (e) {
```

Comment:
I might be mistaken about this, but I believe that `useEffect` runs after every re-render, and that `useFormikContext()` causes a re-render on each change to the form values: i.e. as the user types a value. Catching an exception is fairly expensive, so that could add input latency. I think it might be better to `import { isWebUri } from 'valid-url';` and check that the url is valid before trying to construct a `URL` object from it.

Reply:
From my tests I didn't notice input lag, I would not add another package just for this because it also has its tradeoffs (bigger bundle size), if you do experience lag let's change it.

Reply:
I tried testing for input latency, and the difference was small enough that I'm not confident that it exists. `valid-url` is already among our dependencies, but I don't know if it ends up in the bundle and I suppose it doesn't make a difference.

Reply:
Oh, that package should be explicitly required in `package.json` then. We shouldn't depend on peer deps.